PhysicsTools/PatAlgos/python/slimming/applySubstructure_cfi.py

```diff
@@ -0,0 +1,158 @@
+        btagDiscriminators = ['pfCombinedSecondaryVertexBJetTags', 'pfCombinedInclusiveSecondaryVertexV2BJetTags'],
+        jetCorrections = ('AK4PFchs', ['L1FastJet', 'L2Relative', 'L3Absolute'], 'None'),
+        explicitJTA = True,  # needed for subjet b tagging
+        svClustering = True, # needed for subjet b tagging
+        genJetCollection = cms.InputTag('slimmedGenJets'), 
+        fatJets=cms.InputTag('ak8PFJetsCHS'),             # needed for subjet flavor clustering
+        groomedFatJets=cms.InputTag('ak8PFJetsCHSSoftDrop') # needed for subjet flavor clustering
+    )
+    process.selectedPatJetsAK8PFCHSSoftDrop.cut = cms.string("pt > 200")
+    
+    process.slimmedJetsAK8PFCHSSoftDropSubjets = cms.EDProducer("PATJetSlimmer",
+        src = cms.InputTag("selectedPatJetsAK8PFCHSSoftDropSubjets"),
+        packedPFCandidates = cms.InputTag("packedPFCandidates"),
+        dropJetVars = cms.string("1"),
+        dropDaughters = cms.string("0"),
+        dropTrackRefs = cms.string("1"),
+        dropSpecific = cms.string("1"),
+        dropTagInfos = cms.string("1"),
+    )
+
+    
+    ## Establish references between PATified fat jets and subjets using the BoostedJetMerger
+    process.slimmedJetsAK8PFCHSSoftDropPacked = cms.EDProducer("BoostedJetMerger",
+        jetSrc=cms.InputTag("selectedPatJetsAK8PFCHSSoftDrop"),
+        subjetSrc=cms.InputTag("slimmedJetsAK8PFCHSSoftDropSubjets")
+    )
+    
+    addJetCollection(
+        process,
+        labelName = 'CMSTopTagCHS',
+        jetSource = cms.InputTag('cmsTopTagPFJetsCHS'),
+        jetCorrections = ('AK8PFchs', cms.vstring(['L1FastJet', 'L2Relative', 'L3Absolute']), 'None'),
+        btagDiscriminators = ['pfCombinedSecondaryVertexBJetTags', 'pfCombinedInclusiveSecondaryVertexV2BJetTags'],
+        genJetCollection = cms.InputTag('slimmedGenJetsAK8'), 
+        getJetMCFlavour = False #
+        )
+    process.patJetsCMSTopTagCHS.addTagInfos = True
+    process.patJetsCMSTopTagCHS.tagInfoSources = cms.VInputTag(
+        cms.InputTag('caTopTagInfosPAT')
+        )
+    process.selectedPatJetsCMSTopTagCHS.cut = cms.string("pt > 200")
+
+    addJetCollection(
+        process,
+        labelName = 'CMSTopTagCHSSubjets',
+        jetSource = cms.InputTag('cmsTopTagPFJetsCHS','caTopSubJets'),
+        algo = 'AK',  # needed for subjet flavor clustering
+        rParam = 0.8, # needed for subjet flavor clustering        
+        btagDiscriminators = ['pfCombinedSecondaryVertexBJetTags', 'pfCombinedInclusiveSecondaryVertexV2BJetTags'],
+        jetCorrections = ('AK4PFchs', ['L1FastJet', 'L2Relative', 'L3Absolute'], 'None'),
+        genJetCollection = cms.InputTag('slimmedGenJets'), # Using ak4GenJets for matching which is not entirely appropriate
+        explicitJTA = True,  # needed for subjet b tagging
+        svClustering = True, # needed for subjet b tagging
+        fatJets=cms.InputTag('ak8PFJetsCHS'),             # needed for subjet flavor clustering
```

Comment:
Since ak8 and ca8 jets will generally have different shapes and ghost-association is used to associate hadrons and secondary vertices to subjets, there could be cases with unexpected associations and correspondingly unexpected subjet flavor or b-tagging result. A proper solution would be to remake ca8 jets on the fly.